G2P.pm

```diff
@@ -429,94 +469,189 @@ sub set_variant_include_list_flag {
+    if ($self->variants_filtered_by_frequency_threshold($af_threshold, [$self->{vf_cache_name}])) {
       push @results, "$ar=$zyg";
     }
   }
   return join(',', @results);
 }
 
+=head2 is_g2p_complete
+
+  Arg [1]    : TranscriptVariationAllele $tva
+  Arg [2]    : String $zygosity           
+  Description: A G2P gene is considered complete if its allelic requirement is fulfilled.
+               Create a summary string which connects the fulfilled allelic_requirement and all variants which pass filtering. 
+  Returntype : String $g2p_complete, for example "monoallelic=HET:7_941481_C/T&HET:7_929274_A/G,HOM:7_931481_C/T|biallelic=HOM:7_931481_C/T"
+  Exceptions : None
+  Caller     : General
+  Status     : Stable
+
+=cut
 sub is_g2p_complete {
   my $self = shift;
   my $tva = shift;
   my $zyg = shift;
   my $vf = $tva->base_variation_feature;
   my $individual = $vf->{individual};
   my $transcript = $tva->transcript;
-  my $gene_stable_id = $transcript->{_gene}->stable_id;
+  my $gene_stable_id = get_gene_stable_id($transcript);
   my $transcript_stable_id = $transcript->stable_id; 
   $self->{per_individual}->{$individual}->{$transcript_stable_id}->{$zyg}->{$self->{vf_cache_name}} = 1;
   my @allelic_requirements = keys %{$self->{ar}->{$gene_stable_id}};
-  my $G2P_complete;
+  my @g2p_complete = ();
   foreach my $ar (@allelic_requirements) {
     my $zyg2var = $self->{per_individual}->{$individual}->{$transcript_stable_id};
-    my $fulfils_ar = $self->obeys_rule($ar, $zyg2var);
-    if (scalar keys %$fulfils_ar > 0) {
-      $G2P_complete .= "$ar=";
-      my @passed_variants = ();
-      foreach my $zyg (keys %$fulfils_ar) {
-        my @tmp = ();
-        foreach my $var (@{$fulfils_ar->{$zyg}}) {
-          push @tmp, "$zyg:$var";
-        }
-        push @passed_variants, join('&', @tmp);
+    my $filtered_zyg2var = $self->zyg2var_filtered_by_allelic_requirement_rule($ar, $zyg2var);
+    if (defined $filtered_zyg2var) {
+      my @filtered_variants = ();
+      foreach my $zyg (keys %$filtered_zyg2var) {
+        push @filtered_variants, join('&', map {"$zyg:$_"} @{$filtered_zyg2var->{$zyg}});
       }
-      $G2P_complete = "$ar=" . join(',', @passed_variants);
+      push @g2p_complete, "$ar=" . join(',', @filtered_variants);
     }
   }
-  return $G2P_complete;
+  return join('\|', @g2p_complete);
 } 
 
-sub obeys_rule {
+=head2 zyg2var_filtered_by_allelic_requirement_rule
+
+  Arg [1]    : String $ar allelic requirement
+  Arg [2]    : Hashref $zyg2var
+  Example    : $zyg2var_filtered = $self->zyg2var_filtered_by_allelic_requirement_rule('biallelic', {
+                'HET' => {
+                  '4_32941481_C/T' => 1,
+                  '4_32929274_A/G' => 1
+                }
+               });
+  Description: Check if the variants fulfill the given allelic requirement. Variants are grouped by their
+               zygosity. The subroutine checks for each variant if in internally stored frequency for a variant
```

Comment:
for each variant if in internally stored frequency for a variant -> there's a typo, maybe it should be "if the internally stored frequency for a variant"?